### ISSUE TITLE
Updating from sled store prior to 55772b

### ISSUE THREAD

**Issue by david-perez** [whisperfish/presage#78]
Hi! First of all, thank you for this software. I've been running commit f9ae9b233d3b0119c1dce446f50f7a1416204406 in an application of mine since February without any problems.

Since the root CA changed in https://github.com/whisperfish/libsignal-service-rs/pull/168 and my application stopped working, I figured it would be a good time to update `presage` too.

However, when I now invoke `presage` it tells me that my phone number is not yet registered. I root-caused it to the sled store having changed: in 55772b71b10c8b8dfb8f8e9e92f5339c4d4d7a57 a sled key changed from `state` to `registration`. I've not been keeping up with updates to the project, so perhaps other changes to sled store usage have been made since February. Is the sled store format used by `presage` documented somewhere?

Is it possible to update my sled store to the new format? I can't start afresh by creating the sled store in the new format because the phone number is already registered with Signal, and it appears Signal won't let me re-register it.

**Comment by gferon** [comment]
Unfortunately I broke the store API multiple times this year because of necessary changes, and never really paid extra attention to the backwards compatibility of it as the main use-case that I was looking at was secondary devices (which you can relink as much as you want).

In your case, it should be possible to do something about it, but I'm mostly curious as to why you cannot re-register, which you should always be able to do (perhaps you used a PIN code and need to provide it again?). In theory, re-registering will void your old credentials and keys and issue new ones.

**Comment by david-perez** [comment]
> I'm mostly curious as to why you cannot re-register

I'm not exactly sure. I'm using a landline phone number, so I can't receive SMS. When I attempt to re-register with `--use-voice-call`, an automated voice says "We're sorry, an application error has occurred" and hangs up. I simply attributed this to Signal not wanting to re-register my number.

**Comment by gferon** [comment]
I see. I'll try to register a device with the same version as the one you're using and figure out a way forward ⏩

**Comment by gferon** [comment]
I gave a shot to blindly write a small piece of code that should help you try to migrate your old DB. Can you please clone the branch from https://github.com/whisperfish/presage/tree/migrate-old-presage and `cargo run --example=migration` on a **copy** of your database?

I couldn't test it because I couldn't register with the old `presage`, but there's a good chance it'll work 😄 

When `presage` becomes 0.4, the migration will be automatically applied and supported for every new version, so this never happens again.

**Comment by david-perez** [comment]
I ran the migration tool and got a panic here:

https://github.com/whisperfish/presage/blob/33cabd403a938942fef61c1d0ff91a0592942dbd/src/manager.rs#L493-L497

I changed it to `unwrap_or` and used a value of my own identifying my landline phone and it worked fine (not sure if that could cause problems down the line). I have now been able to update to the latest commit of pesage and remove my custom patches. Thank you!
